src/deepsparse/v2/pipeline.py

```diff
@@ -27,6 +27,7 @@
 from deepsparse.v2.utils import InferenceState, PipelineState
 from deepsparse.v2.utils.data import SubGraph
 from deepsparse.v2.utils.helpers import run_func
+from src.deepsparse.v2.middleware.abstract_middleware import AbstractMiddleware
 
 
 __all__ = ["Pipeline"]
```

Comment:
I think we'll also need to add `start_event` and `end_event` in the `__call__` method of pipeline since it doesn't call into the super method of `Operator`

additionally should think through if there's a way to mark the overall start and end of inference without worrying about nested pipelines